### DISCUSSION TITLE
Sorting by ID

### DISCUSSION THREAD

**Discussion by bridgekat** [isar/isar#543] [Help]
**Question 1:** In using a composite index:

```dart
@Collection()
class Item {
  @Id()
  late int id;

  @Index(composite: [CompositeIndex('secondKey')])
  late int firstKey;

  late int secondKey;
}
```

If I query on the index like `isar.items.where().firstKeyEqualToSecondKeyBetween(0, 0, 100).findAllSync()`, is it safe to assume that the resulting list will be sorted according to `firstKey` first, then by `secondKey`, **and finally by `id`**? (It does so apparently; I just wanted to make sure if this behavior will be stable in future releases!)

**Question 2:** The [documentation](https://isar.dev/queries.html#filters) seems to suggest that the resulting set is sorted by ID as default. Do we still need to use `.anyId()` in the `.where()` clause to select the (implicit) index on ID, in order to guarantee this ordering? What if we want to retrieve the items with largest ID first?

It seems like for version `3.0.0-dev.2` on Windows, using `isar.items.where(sort: Sort.desc).anyId().findAllSync()` gives the intended ordering (by ID in descending order), but if `anyId()` is removed, the resulting set is still sorted by ID, but in ascending order...

**Comment by Maqcel** [comment]
Thank you for this `.anyId()` I was trying to figure out why my code is not working:
```
    final List<IsarMeasurement> list = isarInstance?.txnSync(
          () => isarInstance.isarMeasurements
              .where(sort: Sort.desc)
              **.anyId()**
              .filter()
              .sensorIdEqualTo(sensorId)
              .offset(currentPage * limit + newItemsThreshold)
              .limit(limit)
              .findAllSync(),
```
managed to solve it thanks! But still I think the documentation is not helping in this topic
